core/chains/evm/txmgr/builder.go

```diff
@@ -138,10 +136,10 @@ func NewEvmBroadcaster(
 	listenerConfig txmgrtypes.BroadcasterListenerConfig,
 	keystore KeyStore,
 	txAttemptBuilder TxAttemptBuilder,
-	nonceSyncer NonceSyncer,
 	logger logger.Logger,
 	checkerFactory TransmitCheckerFactory,
 	autoSyncNonce bool,
 ) *Broadcaster {
-	return txmgr.NewBroadcaster(txStore, client, chainConfig, feeConfig, txConfig, listenerConfig, keystore, txAttemptBuilder, nonceSyncer, logger, checkerFactory, autoSyncNonce, evmtypes.GenerateNextNonce)
+	nonceTracker := NewNonceTracker(logger, txStore, client)
```

Comment:
I was actually thinking to move this completely inside Broadcaster since I don't see other callers, but I guess that works too.